libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -326,6 +326,76 @@ TEST(CppInterface, PreprocessingOnlyModeSecondOverridesDisassemblyMode) {
               HasSubstr("void main(){ }"));
 }
 
+TEST(CppInterface, WarningsOnLine) {
+  shaderc::Compiler compiler;
+  shaderc::CompileOptions options;
+  // By default the compiler will emit a warning on line 2 complaining
+  // 'float' as a deprecated attribute in version 130.
+  const std::string kDeprecatedAttributeShader =
+      "#version 130\n"
+      "attribute float x;\n"
+      "void main() {}\n";
+  shaderc::SpvModule result_suppress_warnings_on_line =
+      compiler.CompileGlslToSpv(kDeprecatedAttributeShader,
+                                shaderc_glsl_vertex_shader, options);
+  EXPECT_TRUE(result_suppress_warnings_on_line.GetSuccess());
+  EXPECT_THAT(
+      result_suppress_warnings_on_line.GetErrorMessage().c_str(),
+      HasSubstr(":2: warning: attribute deprecated in version 130; may be "
+                "removed in future release\n"));
+}
+
+TEST(CppInterface, SuppressWarningsOnLine) {
+  shaderc::Compiler compiler;
+  shaderc::CompileOptions options_suppress_warnings;
+  // Sets the compiler to suppress warnings, so that the deprecated attribute
+  // warning won't be emitted.
+  options_suppress_warnings.SetSuppressWarnings();
+  const std::string kDeprecatedAttributeShader =
+      "#version 130\n"
+      "attribute float x;\n"
+      "void main() {}\n";
+  shaderc::SpvModule result_suppress_warnings_on_line =
+      compiler.CompileGlslToSpv(kDeprecatedAttributeShader,
+                                shaderc_glsl_vertex_shader,
+                                options_suppress_warnings);
+  EXPECT_TRUE(result_suppress_warnings_on_line.GetSuccess());
+  EXPECT_STREQ(result_suppress_warnings_on_line.GetErrorMessage().c_str(), "");
```

Comment:
Right.  These are confusing.  In EXPECT_<predicate> macros, the expected thing is the first argument, and the computed thing is the second argument.
In EXPECT_THAT the computed thing is the first arg, and the predicate is spelled out in the second argument.